File: packages/db-operations/data/seed/export/update_frequency.py
"""The service's update frequency for each Fingertips frequency name.

The service stores how often an indicator is updated as one of the values in
UPDATE_FREQUENCIES in @fphd/utils/update-frequency; the exports translate Pholio's
lookup rows to them.
"""

UPDATE_FREQUENCIES = {
    "Annual": "annually",
    "Monthly (not yet in use)": "monthly",
    "Quarterly (not yet in use)": "quarterly",
}


def update_frequency_value(name):
    """The value for a Fingertips frequency name; a name with none stops the export."""
    try:
        return UPDATE_FREQUENCIES[name]
    except KeyError:
        raise ValueError(f"No update frequency value for the Fingertips frequency {name!r}") from None

File: packages/utils/src/update-frequency.ts
/** How often an indicator's data is updated, or that it no longer will be. */
export const UPDATE_FREQUENCIES = [
  'monthly',
  'quarterly',
  'annually',
  'every-2-years',
  'no-fixed-frequency',
  'no-longer-updated',
] as const;

export type UpdateFrequency = (typeof UPDATE_FREQUENCIES)[number];

export const UPDATE_FREQUENCY_LABELS: Readonly<Record<UpdateFrequency, string>> = {
  monthly: 'Monthly',
  quarterly: 'Quarterly',
  annually: 'Annually',
  'every-2-years': 'Every 2 years',
  'no-fixed-frequency': 'No fixed frequency',
  'no-longer-updated': 'This indicator will no longer be updated',
};
